### PR TITLE
Feature/1.5/patch.12 transaction tm involved service branch

### PR DESCRIPTION
**Pull request by lazan**
commit fdcf45ca7c013dc54133fa136d920660f585a9da is not relevant and going to be removed.

The last one is the most important one.

### CODE REVIEW

File: middleware/common/source/service/call/context.cpp [hunk 1]
```diff
@@ -251,12 +254,11 @@ namespace casual
 
 
          reply::Result result;
-         auto prepared = get_reply();
+         auto [ reply, desc] = get_reply();
```
Comment (ahhud) [inline middleware/common/source/service/call/context.cpp]: Perhaps **decriptor** should be used iso **desc** which can be mistaken for _descending_.

**Review by ahhud — APPROVED** [review]

File: middleware/common/source/transaction/context.cpp [hunk 2]
```diff
@@ -279,7 +279,24 @@ namespace casual
                         log::line( log::category::transaction, "involved message: ", message);
 
                         return communication::device::async::call( communication::instance::outbound::transaction::manager::device(), message);
-                     } 
+                     }
+
+                     void send( const transaction::ID& trid, std::vector< strong::resource::id> resources)
+                     {
+                        Trace trace{ "transaction::local::resources::involved::future"};
```
Comment (OlofGullnas) [inline middleware/common/source/transaction/context.cpp]: Incorrect trace argument. Copy&paste from future() above

File: middleware/common/source/transaction/context.cpp [hunk 3]
```diff
@@ -354,6 +371,30 @@ namespace casual
...
+                           // We absolutely know that this is a distributed transaction (non-local gtrid)
+                           auto start_functor = []( auto& trid)
+                           {
+                              return [ &trid]( auto& resource)
+                              {
+                                 return check::Result{ &resource, resource.start( trid, flag::xa::Flag::no_flags)};
+                              };
+                           };
+                           
+                           // we let the TM know aoubt our resources
```
Comment (OlofGullnas) [inline middleware/common/source/transaction/context.cpp]: Nit: Spelling of about

File: middleware/common/source/transaction/context.cpp [hunk 4]
```diff
@@ -226,34 +226,209 @@ namespace casual
...
+
+               namespace resources::end
                {
-                  if( code != decltype( code)::ok)
-                     code::raise::error( code, context);
-               };
+                  template< typename P, typename R>
+                  void invoke( P&& policy, const Transaction& transaction, R& resources)
+                  {
+                     Trace trace{ "transaction::local::resources::start"};
```
Comment (OlofGullnas) [inline middleware/common/source/transaction/context.cpp]: Felaktigt trace argument. Borde vara "transaction::local::resources::end". Påverkar naturligtvis inte funktion. Förvirrar lite vid läsning av loggar.

**Comment by lazan** [comment]
I did a force push, removing the migrate/resume stuff. We might add it later if/when we understand it better.

**Pull request merged**